### ISSUE TITLE
Build is broken for authorization package

### ISSUE THREAD

**Issue by artsf** [docker/docker#32192]
**Description**

Build is broken for Docker Auth package

**Steps to reproduce the issue:**
```
$ go get github.com/docker/docker/pkg/authorization
# github.com/docker/docker/pkg/authorization
github.com/docker/docker/pkg/authorization/plugin.go:108: cannot assign *plugins.Plugin to plugin (type plugingetter.CompatPlugin) in multiple assignment:
	*plugins.Plugin does not implement plugingetter.CompatPlugin (missing BasePath method)
```

**Comment by thaJeztah** [comment]
ping @anusha-ragunathan PTAL

**Comment by aaronlehmann** [comment]
Looks like it doesn't build on OS X because there's no definition of `BasePath` for OS X. In the `pkg/plugins` package there are `plugins_linux.go` and `plugins_windows.go` files where this method is defined. I think `plugins_linux.go` should be renamed to `plugins_unix.go`.

**Comment by artsf** [comment]
Can we please port the fix to 1.13.* ? :)
